### ISSUE TITLE
[BUG]: missing / bad accessibility of rpi-imager 2.0 with screen readers

### ISSUE THREAD

**Issue by dave090679** [raspberrypi/rpi-imager#1250]
### What happened?

When using a screen reader while running rpi-imager, many of the controls are unlabeled. e.g. the check boxes in the "program options" dialog are just read as "check box", there is no label. in the "edit" dialog, the radio buttons are also unlabeed. The contents of the dialog boxes are also not read by screenreaders (tested with NVDA 2025.3)

### Version

2.0.0 (Default)

### What host operating system were you using?

Windows

### Host OS Version

Windows 11 25H2

### Selected OS

trixie 64-bit

### Which Raspberry Pi Device are you using?

Raspberry Pi 5, 500, and Compute Modules 5

### What kind of storage device are you using?

microSD Card in an internal reader

### OS Customisation

- [ ] Yes, I was using OS Customisation when the bug occurred.

### Relevant log output

```shell

```

**Comment by tdewey-rpi** [comment]
2.0 is explicitly aimed at improving accessibility, so this issue is a high-priority one.

I'd not previously come across NVDA (indeed, using just the acronym, you get Nvidia's stock ticker...). I'll have a look at how I can use this to inspect Imager's labelling.

**Comment by tdewey-rpi** [comment]
Testing with VoiceOver on macOS, I can hear the scale of the issue as reported. To my large surprise, the built-in macOS screen reader does not, in fact, read the screen. Instead, it requires explicit accesibility hints to be attached to every control.

This will be a large change, but the first pass of adding these explicit hints will be present in the final 2.0.0 release. Testing will be conducted by myself using macOS' screen reader, but I think it's a fairly safe assumption that NVDA will also pick up on these changes.

**Comment by Isaac-Leonard** [comment]
I'm not sure if it's just me, but after clicking any of the buttons on the first screen, my screen reader doesn't read out anything else on the screen.
All I can do is click escape to get back to the first screen.
I'm using voiceover on MacOS

**Comment by tdewey-rpi** [comment]
The released 2.0.0 should be vastly improved over the -rc4 build, and I would absolutely appreciate a re-test by other listeners.
